plugins/javadoc/src/main/kotlin/org/jetbrains/dokka/javadoc/pages/htmlPreprocessors.kt

```diff
@@ -90,4 +96,61 @@ object IndexGenerator: PageTransformer {
             IndexPage(i + 1, set.sortedBy { it.getId().toLowerCase() }, keys, input.sourceSets())
         })
     }
+}
+
+object DeprecatedPageCreator : PageTransformer {
+    override fun invoke(input: RootPageNode): RootPageNode {
+        val elements = IdentityHashMap<DeprecatedPageSection, MutableSet<DeprecatedNode>>().apply {
```

Comment:
If `DeprecatedPageSection` is made into enum this identity map can be just an ordinary map.